--- /home/seb/workspace-hive/robot/wxpython/ext/wxWidgets/src/gtk/dataview2.cpp	2021-04-06 03:04:23.204000000 +0200
+++ /home/seb/workspace-hive/robot/wxpython/ext/wxWidgets/src/gtk/dataview.cpp	2021-04-06 22:13:16.610903764 +0200
@@ -343,7 +343,7 @@
         }
     }
 
-    void AddNode( wxGtkTreeModelNode* child )
+    void AddNode( wxGtkTreeModelNode* child, bool allow_sort=true )
         {
             m_nodes.Add( child );
 
@@ -351,7 +351,7 @@
 
             m_children.Add( id );
 
-            if (m_internal->ShouldBeSorted())
+            if (allow_sort && m_internal->ShouldBeSorted())
             {
                 gs_internal = m_internal;
                 m_children.Sort( &wxGtkTreeModelChildCmp );
@@ -389,16 +389,16 @@
             m_children.Insert( id, pos );
         }
 
-    void AddLeaf( void* id )
+    void AddLeaf( void* id, bool allow_sort=true )
         {
-            InsertLeaf(id, m_children.size());
+            InsertLeaf(id, m_children.size(), allow_sort);
         }
 
-    void InsertLeaf( void* id, unsigned pos )
+    void InsertLeaf( void* id, unsigned pos, bool allow_sort=true )
         {
             m_children.Insert( id, pos );
 
-            if (m_internal->ShouldBeSorted())
+            if (allow_sort && m_internal->ShouldBeSorted())
             {
                 gs_internal = m_internal;
                 m_children.Sort( &wxGtkTreeModelChildCmp );
@@ -3786,12 +3785,13 @@
         unsigned int pos;
         for (pos = 0; pos < count; pos++)
         {
+        	bool allow_sort = (pos==count-1) ;
             wxDataViewItem child = children[pos];
 
             if (m_wx_model->IsContainer( child ))
-                node->AddNode( new wxGtkTreeModelNode( node, child, this ) );
+                node->AddNode( new wxGtkTreeModelNode( node, child, this ), allow_sort );
             else
-                node->AddLeaf( child.GetID() );
+                node->AddLeaf( child.GetID(), allow_sort );
 
             // Don't send any events here
         }
